rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/monitoring/views/compact/Tile.java

```diff
@@ -0,0 +1,333 @@
+
+                MenuItem deployItem = new MenuItem("Deploy", deployItemImageResource);
+                deployItem.addClickHandler(event12 -> compactView.getController().deployNodeSource());
+
+                MenuItem undeployItem = new MenuItem("Undeploy", undeployItemImageResource);
+                undeployItem.addClickHandler(event1 -> compactView.getController().undeployNodeSource());
+
+                MenuItem editItem = new MenuItem("Edit", editItemImageResource);
+                String nodeSourceName = nodesource == null ? "" : nodesource.getSourceName();
+                editItem.addClickHandler(event1 -> compactView.getController().editNodeSource(nodeSourceName));
+
+                if (node != null) {
+                    if (node.isLocked()) {
+                        lockItem.setEnabled(false);
+                        unlockItem.setEnabled(true);
+                    } else {
+                        lockItem.setEnabled(true);
+                        unlockItem.setEnabled(false);
+                    }
+                }
+
+                if (nodesource != null) {
+                    switch (nodesource.getNodeSourceStatus()) {
+                        case NODES_DEPLOYED:
+                            enableItems(new MenuItem[] { undeployItem });
+                            disableItems(new MenuItem[] { deployItem, editItem });
+                            break;
+                        case NODES_UNDEPLOYED:
+                            enableItems(new MenuItem[] { deployItem, editItem });
+                            disableItems(new MenuItem[] { undeployItem });
+                            break;
+                        default:
+                            disableItems(new MenuItem[] { deployItem, undeployItem, editItem });
+                    }
+                } else {
+                    disableItems(new MenuItem[] { deployItem, undeployItem, editItem });
+                }
+
+                menu.setItems(deployItem,
+                        undeployItem,
+                        editItem,
+                        new MenuItemSeparator(),
+                        lockItem,
+                        unlockItem,
+                        removeItem);
+
+                menu.moveTo(event.getClientX(), event.getClientY());
+                menu.show();
+
+                event.stopPropagation();
+                event.preventDefault();
+                break;
+            default:
+                super.onBrowserEvent(event);
+                break;
+        }
+    }
+
+    private void disableItems(MenuItem[] items) {
+        for (MenuItem item : items) {
+            item.setEnabled(false);
+        }
+    }
+
+    private void enableItems(MenuItem[] items) {
+        for (MenuItem item : items) {
+            item.setEnabled(true);
+        }
+    }
+
+    private void init() {
+        setSize("16px", "16px");
+        getElement().getStyle().setBorderColor("white");
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "white" 6 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)